tests/Type/data/eloquent-builder.php

```diff
@@ -432,6 +432,11 @@ function testPaginateItems()
     assertType('array<App\User>', User::query()->paginate()->items());
 }
 
+function testChunkOnEloquentBuilder()
+{
+    User::chunk(1000, fn ($collection) => assertType('Illuminate\Database\Eloquent\Collection<int, App\User>'));
```

Comment:
I'm not a developer but after reading the docblock
```suggestion
    assertType('Illuminate\Database\Eloquent\Collection<int, App\User>', User::chunk(1000, fn ($collection, $someInteger) => $someBoolean));
```